pkg/cmd/server/origin/leaderelection.go

```diff
@@ -81,10 +87,6 @@ func NewLeaderElection(options configapi.MasterConfig, leader componentconfig.Le
 
 	// use the endpoints leader election path.
 	plug := plug.New(false)
-	events := record.NewBroadcaster()
-	events.StartLogging(glog.Infof)
-	events.StartRecordingToSink(&v1core.EventSinkImpl{Interface: v1core.New(kc.Core().RESTClient()).Events("")})
-	lock.LockConfig.EventRecorder = events.NewRecorder(kapi.Scheme, v1corev1.EventSource{Component: name})
 	elector, err := leaderelection.NewLeaderElector(leaderelection.LeaderElectionConfig{
```

Comment:
Also, I assume `legacyLeaderElectionStart` would need all possible lock types since either an endpoint or config map lock should make it exit.